extensions/common/vault/vault-azure/src/main/java/org/eclipse/edc/vault/azure/AzureVaultExtension.java

```diff
@@ -22,14 +22,11 @@
 import org.eclipse.edc.runtime.metamodel.annotation.Setting;
 import org.eclipse.edc.spi.monitor.Monitor;
 import org.eclipse.edc.spi.security.CertificateResolver;
-import org.eclipse.edc.spi.security.PrivateKeyResolver;
 import org.eclipse.edc.spi.security.Vault;
-import org.eclipse.edc.spi.security.VaultCertificateResolver;
-import org.eclipse.edc.spi.security.VaultPrivateKeyResolver;
 import org.eclipse.edc.spi.system.ServiceExtension;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
 
-@Provides({Vault.class, PrivateKeyResolver.class, CertificateResolver.class})
+@Provides({Vault.class, CertificateResolver.class})
```

Comment:
it does not provide a `CertificateResolver` anymore

Reply:
refactored the entire extension and the test a bit.

Reply:
I meant, the `CertificateResolver.class` could be removed by the `@Provides` annotation right?

Reply:
yep, same for the `PrivateKeyResolver`. but at that point, the only that gets created is the vault, so I opted to use the `@Provider` pattern. 
Consequently, the test needed some love too.

Reply:
but in this PR the `@Provider` is not used, and the `@Provides` is, maybe you are confusing with another PR

Reply:
ha, no actually I already had a branch `fix/compile_errors` locally, which did not track the upstream branch (I created the original PR on another machine). 
OFC I pushed into that completely unassociated branch 🙄